pol-core/uoconvert/UoConvertMain.cpp

```diff
@@ -765,31 +765,72 @@ void UoConvertMain::ProcessSolidBlock( unsigned short x_base, unsigned short y_b
   mapwriter.SetSolidx2Offset( x_base, y_base, idx2_offset );
 }
 
-void UoConvertMain::write_multi( FILE* multis_cfg, unsigned id, FILE* multi_mul,
-                                 unsigned int offset, unsigned int length )
+std::string UoConvertMain::resolve_type_from_id( unsigned id ) const
 {
-  USTRUCT_MULTI_ELEMENT elem;
-  unsigned int count;
-  if ( cfg_use_new_hsa_format )
-    count = length / sizeof( USTRUCT_MULTI_ELEMENT_HSA );
+  if ( BoatTypes.count( id ) )
+    return "Boat";
   else
-    count = length / sizeof elem;
+    return "Multi";
+}
 
-  std::string type, mytype;
-  if ( BoatTypes.count( id ) )
-    type = "Boat";
-  else if ( HouseTypes.count( id ) )
-    type = "House";
-  else if ( StairTypes.count( id ) )
-    type = "Stairs";
+void UoConvertMain::write_multi_element( FILE* multis_cfg, const USTRUCT_MULTI_ELEMENT& elem,
+                                         const std::string& mytype, bool& first )
+{
+  if ( elem.graphic == GRAPHIC_NODRAW )
+    return;
+
+  std::string type = elem.flags ? "static" : "dynamic";
```

Comment:
elem.flags is checked here directly - shouldn't that be changed as well?

Reply:
i don't think so? since the uop -> mul just stores 0 for dynamic, 1 for static